### PR TITLE
SVCPLAN-5555: Add support for SUSE

### PR DESCRIPTION
**Pull request by billglick**
This is being tested on `control-test-opensuse155a`.

### CODE REVIEW

**Comment by jakerundall** [comment]
This seems fine and makes a working config on the openSUSE test node as far as I can see.

But do we not need to copy over all of the other firewall chain names from https://github.com/ncsa/puppet-profile_firewall/blob/main/data/os/RedHat.yaml ?

**Comment by billglick** [comment]
@jakerundall I was thinking the lookup options on `profile_firewall::inbuilt_chains` to merge in `data/common` made the duplicate rules redundant since they are also referenced in `data/common` already. But perhaps those lookup options need to be adjusted to be `deep`?

@bsper2 Do you have a trick for identifying what the default `inbuilt_chains` are for an OS version?

**Comment by jakerundall** [comment]
> @jakerundall I was thinking the lookup options on `profile_firewall::inbuilt_chains` to merge in `data/common` made the duplicate rules redundant since they are also referenced in `data/common` already. But perhaps those lookup options need to be adjusted to be `deep`?

Got it, I missed that.

**Comment by bsper2** [comment]
> @bsper2 Do you have a trick for identifying what the default inbuilt_chains are for an OS version?

Yeah, there are 5 tables : filter, nat, mangle, raw, security

So for each table do a : iptables -t TABLE -S
And grab the chain names from the output of each of those

**Comment by bsper2** [comment]
Oh, and also the rules might be different between ipv4/ipv6 (although diffs there are unlikely) but there are also rules down on the ethernet level which you can print the chains names with:

ebtables -t TABLE-NAME -L

ebtables won't have all the 5 tables, I think it only supports nat and filter since it works at Layer 2

**Comment by billglick** [comment]
Here is what I'm doing to identify the inbuilt chains:
```bash
IPTABLES="filter nat mangle raw security"
for table in $IPTABLES; do echo "iptables Table ${table} rules:"; iptables -t $table -S; echo ; done
for table in $IPTABLES; do echo "ip6tables Table ${table} rules:"; ip6tables -t $table -S; echo ; done

EBTABLES="filter nat broute"
for table in $EBTABLES; do echo "Ethernet Bridge Table ${table} rules:"; ebtables -t $table -L; echo ; done
```

**Comment by billglick** [comment]
`iptables` under SUSE/SLES seems odd. 
- There does not seem to be any `iptables` services that `systemctl` knows of. The [puppetlabs/firewall](https://github.com/puppetlabs/puppetlabs-firewall) Puppet module seems to not know how to deal with it as a service either -- see https://github.com/puppetlabs/puppetlabs-firewall/blob/810e7fac43d0e752378be56419a87d6339c0aedf/manifests/params.pp#L100 .
- And the iptables rules we are creating do not seem to persist on reboot. They are getting recreated on the initial Puppet agent runs after a reboot.
- There is an old issue about SLES/SUSE support that was closed but never seemed to be addressed: https://github.com/puppetlabs/puppetlabs-firewall/issues/180 . So despite the module saying it supports SLES, I'm not sure how well it supports it.

File: data/os/RedHat.yaml [hunk 1]
```diff
@@ -1,26 +1,26 @@
 ---
 profile_firewall::inbuilt_chains:
```
Comment (billglick) [inline data/os/RedHat.yaml:2]: I audited and fixed the ordering of these. Nothing was technically wrong with the ordering, but I sorted each group of tables alphabetically.

File: data/common.yaml [hunk 2]
```diff
@@ -14,27 +14,27 @@ profile_firewall::ignores: {}
 # Leave some generic inbuilt_chains, but each OS should have their own
 # copy of inbuilt_chains in hiera under data/os/..
 profile_firewall::inbuilt_chains:
```
Comment (billglick) [inline data/common.yaml:16]: I audited and fixed the ordering of these. Nothing was technically wrong with the ordering, but I sorted each group of tables alphabetically.

**Review by jakerundall — APPROVED** [review]

**Pull request merged**